src/components/CCData.js

```diff
@@ -32,8 +31,33 @@ class CCData extends Component {
             knownStates: false,
             errorMessage: ''
         };
+        if(Object.keys(this.props.currentCCInfo)){
+            Object.keys(this.props.currentCCInfo).forEach((key) => {
+                this.state.ccinfo[key] = this.props.currentCCInfo[key];
```

Comment:
why not replace this with "this.state.ccinfo = props.currentCCInfo"?

Reply:
It comes in as an empty object first time, so that would override existing and cause problems. If the file happens to come in incomplete, original structure from the constructor stays. Mainly protection of structure and declarations